src/core/dbus-execute.c

```diff
@@ -2027,21 +2042,37 @@ int bus_exec_context_set_transient_property(
         }
 
         if (streq(name, "ProtectHostnameEx")) {
-                const char *s;
-                ProtectHostname t;
+                const char *s, *h = NULL;
 
-                r = sd_bus_message_read(message, "s", &s);
+                r = sd_bus_message_read(message, "(ss)", &s, &h);
                 if (r < 0)
                         return r;
 
-                t = protect_hostname_from_string(s);
-                if (t < 0)
+                if (!isempty(h)) {
+                        if (!hostname_is_valid(h, /* flags = */ 0))
+                                return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid hostname in %s setting: %s", name, h);
+
+                        r = dns_name_is_valid(h);
```

Comment:
any reason you validate the argument by DNS rules too here? seems unnecessary to me. We don't do that on SetHostname() in hostnamed either, and I think we should keep the rules in sync on this.

Reply:
The conf parser was introduced by a7d0ef446e5f28b37688743de22052abc8136a94.

Reply:
The check was introduced by 6528693a94821e0e13a8e112fb481c4ab0c62688, to check the specified hostname can be used by DHCP clients. So, actually the check is not necessary here. I will drop the check.